### PR TITLE
Switch from package syscall to golang.org/x/sys/unix

### PR DESCRIPTION
**Pull request by tklauser**
**What this PR does / why we need it**:

The syscall package is locked down and the comment in https://github.com/golang/go/blob/master/src/syscall/syscall.go#L21-L24 advises to switch code to use the corresponding package from golang.org/x/sys. This PR does so and replaces usage of package syscall with package golang.org/x/sys/unix where applicable. This will also allow to get updates and fixes
without having to use a new go version.

In order to get the latest functionality, golang.org/x/sys/ is re-vendored. This also allows to use Eventfd() from this package instead of calling the eventfd() C function.

**Special notes for your reviewer**:

This follows previous works in other Go projects, see e.g. moby/moby#33399, cilium/cilium#588

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/rollback.go [hunk 1]
```diff
@@ -108,7 +108,7 @@ func (r *DeploymentRollbacker) Rollback(obj runtime.Object, updatedAnnotations m
 // watchRollbackEvent watches for rollback events and returns rollback result
 func watchRollbackEvent(w watch.Interface) string {
 	signals := make(chan os.Signal, 1)
-	signal.Notify(signals, os.Interrupt, os.Kill, syscall.SIGTERM)
+	signal.Notify(signals, os.Interrupt, os.Kill, unix.SIGTERM)
```
Comment (resouer) [inline pkg/kubectl/rollback.go]: May I ask what's the go version required for this?

Comment (tklauser) [inline pkg/kubectl/rollback.go]: I have only tested this with go 1.8.3 but I'd guess this should work with every version, as the underlying type of both `syscall.Signal` and `unix.Signal` is `int`.

**Comment by tklauser** [comment]
The federation-e2e-gce test failure seems unrelated to the PR as indicated by the following log messages in https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/49300/pull-kubernetes-federation-e2e-gce/16070/?log#log:

```
...
I0720 14:57:30.326] Your "OAuth 2.0 Service Account" credentials are invalid. Please run
I0720 14:57:30.326]   $ gcloud auth login
I0720 14:57:30.326] Duplicate type [0:0:4]
I0720 14:57:43.091] CommandException: 1 file/object could not be transferred.
I0720 14:57:43.181] FAILED
I0720 14:57:43.184] 
I0720 14:57:43.187] 
I0720 14:57:43.189] Signal ERR caught!
I0720 14:57:43.192] 
I0720 14:57:43.194] Traceback (line function script):
I0720 14:57:43.195] 192 main /go/src/k8s.io/release/push-build.sh
I0720 14:57:43.197] 
I0720 14:57:43.200] Exiting...
I0720 14:57:43.202] 
I0720 14:57:43.205] 
I0720 14:57:43.206] push-build.sh: DONE main on ed611d83e0bd Thu Jul 20 14:57:43 UTC 2017 in 41s
W0720 14:57:43.306] 2017/07/20 14:57:43 util.go:131: Step '/go/src/k8s.io/release/push-build.sh --nomock --verbose --noupdatelatest --bucket=kubernetes-release-pull --ci --gcs-suffix=/pull-kubernetes-federation-e2e-gce --federation' finished in 40.529395722s
W0720 14:57:43.307] 2017/07/20 14:57:43 main.go:182: Saved XML output to /workspace/_artifacts/junit_runner.xml.
W0720 14:57:43.307] 2017/07/20 14:57:43 main.go:220: Something went wrong: failed to acquire k8s binaries: error during /go/src/k8s.io/release/push-build.sh --nomock --verbose --noupdatelatest --bucket=kubernetes-release-pull --ci --gcs-suffix=/pull-kubernetes-federation-e2e-gce --federation: exit status 2
...
```

File: pkg/kubectl/rollback.go (hunk 1 above)
Comment (luxas) [inline pkg/kubectl/rollback.go]: @resouer We only support go1.8.3 in any case...

**Comment by cmluciano** [comment]
Should these files be guarded with build tags?

**Comment by tklauser** [comment]
@cmluciano AFAICT all files touched by this PR - except `pkg/kubectl/rollback.go` - already have either build tags or the filename set accordingly. As for `rollback.go` I'm not sure now whether this should even be converted or whether this is an arch-independent file? In the latter case, switching to x/sys/unix wouldn't be valid. Any advise there?

**Comment by luxas** [comment]
/test pull-kubernetes-cross

will tell us whether this works on all platforms

**Comment by tklauser** [comment]
```
I0721 10:00:32.022] +++ [0721 09:58:21] windows/amd64: go build started
I0721 10:00:32.023] # k8s.io/kubernetes/pkg/kubectl
I0721 10:00:32.023] pkg/kubectl/rollback.go:111: undefined: unix.SIGTERM
```

Apparently pkg/kubectl/rollback.go is also needed for Windows, so `syscall.SIGTERM` needs to stay there. I just pushed an updated version of the PR.

**Comment by tklauser** [comment]
kubemark-e2e-gce failure seems to be the same as #49320

**Comment by luxas** [comment]
Test the cross-build again:
/test pull-kubernetes-cross

For the pull-kubernetes-e2e-gce-etcd3 flake:
/retest

**Comment by thockin** [comment]
This needs an associated issue and a `closes #12345` in the FIRST COMMENT, or it can not be approved.

/lgtm

**Comment by tklauser** [comment]
Seems like the results for pull-kubernetes-cross and @k8s-ci-robot
pull-kubernetes-kubemark-e2e-gce weren't picked up, even though they were successful. Maybe due to the Github outage yesterday?

**Comment by tklauser** [comment]
There still seems to be some issue communicating back the status of pull-kubernetes-kubemark-e2-gce :(

/retest

**Comment by fejta** [comment]
I believe the hung status was a prow/plank bug that was fixed earlier this week

/test pull-kubernetes-kubemark-e2e-gce

**Comment by tklauser** [comment]
Thank you @fejta!

**Pull request merged**